pykafka/simpleconsumer.py

```diff
@@ -462,6 +477,15 @@ def _handle_success(parts):
                     [part for errcode, parts in parts_by_error.iteritems()
                      for part in parts])
 
+            # release all locks to allow fetching
+            for errcode, owned_partitions in parts_by_error.iteritems():
```

Comment:
I see what you mean, when `reset_offsets` is called internally by the error handler that's a bit messy. I was mostly thinking of use of `reset_offsets` by the user application - if we don't raise an exception, does that mean the user must always check offsets on messages to see if any errors occured?